src/iOS/Xamarin.iOS/Samples/Layers/ChangeStretchRenderer/ChangeStretchRenderer.cs

```diff
@@ -62,374 +51,282 @@ public override void ViewDidLoad()
+            StretchRenderer stretchRenderer = null;
 
-            // Create a color ramp for the stretch renderer
-            ColorRamp myColorRamp = ColorRamp.Create(PresetColorRampType.DemLight, 1000);
-
-            // Create the place holder for the stretch renderer
-            StretchRenderer myStretchRenderer = null;
-
-            switch (myRendererTypeChoice)
+            switch (_rendererTypes.SelectedSegment)
             {
-                case "Min Max":
+                case 0:
 
-                    // This section creates a stretch renderer based on a MinMaxStretchParameters
-                    // TODO: Add you own logic to ensure that accurate min/max stretch values are used
+                    // This section creates a stretch renderer based on a MinMaxStretchParameters.
+                    // TODO: Add you own logic to ensure that accurate min/max stretch values are used.
 
-                    // Create an IEnumerable from a list of double min stretch value doubles
-                    IEnumerable<double> myMinValues = new List<double> { Convert.ToDouble(_Input_Parameter1.Text) };
+                    try
+                    {
+                        // Create an IEnumerable from a list of double min stretch value doubles.
+                        IEnumerable<double> minValues = new List<double> { Convert.ToDouble(_inputParameter1.Text) };
 
-                    // Create an IEnumerable from a list of double max stretch value doubles
-                    IEnumerable<double> myMaxValues = new List<double> { Convert.ToDouble(_Input_Parameter2.Text) };
+                        // Create an IEnumerable from a list of double max stretch value doubles.
+                        IEnumerable<double> maxValues = new List<double> { Convert.ToDouble(_inputParameter2.Text) };
 
-                    // Create a new MinMaxStretchParameters based on the user choice for min and max stretch values
-                    MinMaxStretchParameters myMinMaxStretchParameters = new MinMaxStretchParameters(myMinValues, myMaxValues);
+                        // Create a new MinMaxStretchParameters based on the user choice for min and max stretch values.
+                        MinMaxStretchParameters minMaxStretchParameters = new MinMaxStretchParameters(minValues, maxValues);
 
-                    // Create the stretch renderer based on the user defined min/max stretch values, empty gamma values, statistic estimates, and a predefined color ramp 
-                    myStretchRenderer = new StretchRenderer(myMinMaxStretchParameters, myGammaValues, true, myColorRamp);
+                        // Create the stretch renderer based on the user defined min/max stretch values, empty gamma values, statistic estimates, and a predefined color ramp.
+                        stretchRenderer = new StretchRenderer(minMaxStretchParameters, gammaValues, true, colorRamp);
+                    }
+                    catch (Exception ex)
```

Comment:
If we're just guarding against `Convert.ToDouble` exceptions, this can be narrowed to catching ArgumentExceptions.

Also, variable `ex` is not used and can be omitted.